File: src/Designer/backend/src/Designer/Models/Metrics/InstanceErrorMetric.cs
using System;

namespace Altinn.Studio.Designer.Models.Metrics;

public class InstanceErrorMetric
{
    public required string Name { get; set; }
    public required string InstanceId { get; set; }
    public required double Count { get; set; }
    public required Uri LogsUrl { get; set; }
}

File: src/Designer/frontend/admin/features/apps/types/metrics/InstanceErrorMetric.ts
export type InstanceErrorMetric = {
  name: string;
  instanceId: string;
  count: number;
  logsUrl: string;
};

// Error categories that target a specific instance and can therefore be broken down per instance.
const instanceScopedErrorMetrics = ['failed_process_next_requests'];

export const isInstanceScopedErrorMetric = (name: string): boolean =>
  instanceScopedErrorMetrics.includes(name);
